redfish-core/lib/log_services.hpp

```diff
@@ -6720,27 +6720,138 @@ inline void getRedfishUriByDbusObjPath(
+                                                           msgReg->message);
+
+    asyncResp->res.jsonValue[msgPropPath] = msg;
+    asyncResp->res.jsonValue[msgArgPropPath] = messageArgs;
+}
+
+/**
+ * @brief Read the Pretty Name property using the dbus call and load the message
+ * property
+ *
+ * @param[in,out] asyncResp Async response object
+ * @param[in]   path        D-bus object path to find the pretty name
+ * @param[in]   services    List of services to exporting the D-bus object path
+ * @param[in] entryJsonIdx  The json entry index to add isolated hardware
+ *                          details in the appropriate entry json object.
+ * @param[in]   guardType   The guardtype value as a string. To be used in
+ *                          the redfish message property
+ *
+ * @return void
+ */
+inline void updateHwIsolationMessage(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& path, const dbus::utility::MapperServiceMap& services,
+    const size_t entryJsonIdx, const std::string& guardType)
+{
+    // Ensure we only got one service back
+    if (services.size() != 1)
+    {
+        BMCWEB_LOG_ERROR("Invalid Service Size {}", services.size());
+        for (const auto& service : services)
+        {
+            BMCWEB_LOG_ERROR("Invalid Service Name: {}", service.first);
+        }
+        if (asyncResp)
+        {
+            messages::internalError(asyncResp->res);
+        }
+        return;
+    }
+    sdbusplus::asio::getProperty<std::string>(
+        *crow::connections::systemBus, services[0].first, path,
+        "xyz.openbmc_project.Inventory.Item", "PrettyName",
+        [asyncResp, path, entryJsonIdx,
+         guardType](const boost::system::error_code& ec,
+                    const std::string& prettyName) {
+        loadHwIsolationMessage(ec, prettyName, asyncResp, path, entryJsonIdx,
+                               guardType);
+    });
+}
+
+/**
+ * @brief API used to get "PrettyName" by using the given dbus object path
+ *        and fill into "Message" property of LogEntry schema.
  *
- * @return The redfish response with "Message" property of LogEntry schema
- *         if success else nothing in redfish response.
+ * @param[in] asyncResp .   The redfish response to return.
+ * @param[in] dbusObjPath   The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx  The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in]   guardType   The guardtype value as a string. To be used in
+ *                          the redfish message property
+ * @return void
  */
 
 inline void getPrettyNameByDbusObjPath(
     const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     const sdbusplus::message::object_path& dbusObjPath,
-    const size_t entryJsonIdx)
+    const size_t entryJsonIdx, const std::string& guardType)
 {
     crow::connections::systemBus->async_method_call(
```

Comment:
nit: Prefer to use dbus_utility:: getDbusObject()

Reply:
done